sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java

```diff
@@ -203,7 +213,20 @@ public Long getOffset() {
      *     Hubs service or there was no partition key set when the event was sent to the Event Hub.
      */
     public String getPartitionKey() {
-        return systemProperties.getPartitionKey();
+        return (String) amqpAnnotatedMessage.getMessageAnnotations().get(PARTITION_KEY_ANNOTATION_NAME.getValue());
+    }
+
+    /**
+     *  Sets the instant, in UTC, of when the event was enqueued in the Event Hub partition.
+     *
+     * @param enqueuedTime Enqueued time of this message
+     *
+     * @return The updated {@link EventData}.
+     * @see #getEnqueuedTime()
+     */
+    public EventData setEnqueuedTime(Instant enqueuedTime) {
+        amqpAnnotatedMessage.getMessageAnnotations().put(ENQUEUED_TIME_UTC_ANNOTATION_NAME.getValue(), enqueuedTime);
```

Comment:
This should not be settable, it is a system set property. Please do not add properties that are out of scope for this issue: https://github.com/Azure/azure-sdk-for-java/issues/21717

Reply:
According to your comment, remove unnecessary parameters, variables and methods in the new version